aiosmtpd/docs/smtpd.rst

```diff
@@ -0,0 +1,52 @@
+.. _smtpd:
+
+==================================
+ Migrating from smtpd to aiosmtpd
+==================================
+
+aiosmtpd is designed to make it easy to migrate an existing application based
+on smtpd to aiosmtpd.
+
+Consider the following subclass of ``smtpd.SMTPServer``::
+
+    class CustomSMTPServer(smtpd.SMTPServer):
+        def process_message(self, peer, mail_from, rcpt_tos, data):
+            # Process message data...
+            if error_occurred:
+                return '500 Could not process your message'
+
+    if __name__ == '__main__':
+        server = CustomSMTPServer(('127.0.0.1', 10025), None)
+        asyncore.loop()  # Run event loop in current thread
+
+To switch this application to using ``aiosmtpd``, implement a handler with
+the ``handle_DATA()`` method::
+
+    class CustomHandler:
+        async def handle_DATA(self, server, session, envelope):
```

Comment:
I wish we could use `async def` but see #16 which I think for now means we have to document using `@asyncio.coroutine` instead.